### DISCUSSION TITLE
Create host group with assignment rule in multi tenant environment

### DISCUSSION THREAD

**Discussion by bgbg12** [CrowdStrike/falconpy#771] [Q&A] [answered]
Hi,
Has anybody built a script that create host group with assignment rule **in every tenant** in the environment? 

Thanks in advance :)

**Comment by jshcodes** [comment]
I believe I have a variation of this.  Let me clean it up and I'll post it as a sample.  Will add the link here once it's available.

**Reply by bgbg12** [reply]
Great thanks

**Reply by felixguerrero12** [reply]
why not enable member_cid to have multiple tenant id? Adding the member_cid to have multiple tenant and do async requests in the backend.

**Comment by jshcodes** [comment, marked as answer]
Hi @bgbg12 -

The sample is posted [here](https://github.com/CrowdStrike/falconpy/blob/main/samples/flight_control/host_group_duplicator.py). It should duplicate all groups that match the filter you provide using the `-f` argument (wildcard match). Let us know if you have any questions!  :smile:

> I did not implement threading for this sample, but @felixguerrero12 is correct. You could easily make the group creation operations occur asynchronously.

**Reply by bgbg12** [reply]
Hi @jshcodes,
Looks like an excellent solution, thank you very much!
I would appreciate it if you could tell me how to use this sample (if you have a READ ME or something like that) and what permissions I need to get in the API for it to work.
thanks again :)

**Reply by bgbg12** [reply]
@felixguerrero12 @jshcodes

**Reply by jshcodes** [reply]
We'll get a README posted for this shortly.
